inference-engine/src/vpu/graph_transformer/src/middleend/passes/reshape_conv.cpp

```diff
@@ -0,0 +1,104 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+// This is the pass that finds the 3 pattern Convolution stage,
+// Pass adds reshape stage for tensors before and after the Convolution stage,
+// Tiling for new tensors (after reshape) works better
+// and the last one is Relu.
+
+#include <vpu/middleend/pass_manager.hpp>
+
+namespace vpu {
+
+namespace {
+
+class PassImpl final : public Pass {
+public:
+    explicit PassImpl(const StageBuilder::Ptr& stageBuilder) : _stageBuilder(stageBuilder) {}
+
+    void run(const Model& model) override;
+
+private:
+    StageBuilder::Ptr _stageBuilder;
+};
+
+void PassImpl::run(const Model& model) {
+    VPU_PROFILE(hwConvTiling);
+    for (const auto& stage : model->getStages()) {
+        if (stage->type() != StageType::StubConv) {
+            continue;
+        }
+
+        const auto tryHW = stage->attrs().getOrDefault<bool>("tryHW", false);
+        if (!tryHW) {
+            continue;
+        }
+
+        if (stage->attrs().get<int>("kernelSizeX") != 1 ||
+            stage->attrs().get<int>("kernelSizeY") != 1) {
+            continue;
+        }
+
+        const auto input = stage->input(0);
+        const auto output = stage->output(0);
+
+        const auto& inputDesc = input->desc();
+        const auto& outputDesc = output->desc();
+
+        if ((inputDesc.dimsOrder() != DimsOrder::NCHW) ||
+            (outputDesc.dimsOrder() != DimsOrder::NCHW)) {
+            continue;
+        }
+
+        const auto actualInputShape = inputDesc.dims();
+        const auto expectedInputShape = DimValues{ {Dim::N, 1}, {Dim::C, 608}, {Dim::H, 34}, {Dim::W, 60} };
+        std::vector<size_t> outChannels{10, 128, 490};
```

Comment:
Nitpick: declare variables closer to first place where that are used

Reply:
fixed